### PR TITLE
Allow custom cookie name for Ad CID.

### PR DESCRIPTION
**Pull request by lannka**
@rudygalfi please review the description part.

For #8847

### CODE REVIEW

File: extensions/amp-analytics/0.1/cid-impl.js [hunk 1]
```diff
@@ -110,17 +110,14 @@ export class Cid {
...
-    if (typeof externalCidScope == 'string') {
-      getCidStruct = {scope: externalCidScope};
-    } else {
-      getCidStruct = /** @type {!GetCidDef} */ (externalCidScope);
-    }
-    user().assert(/^[a-zA-Z0-9-_.]+$/.test(getCidStruct.scope),
-        'The client id name must only use the characters ' +
-        '[a-zA-Z0-9-_.]+\nInstead found: %s', getCidStruct.scope);
+  get(getCidStruct, consent, opt_persistenceConsent) {
+    dev().assert(typeof getCidStruct != 'string', ''); // DO-NOT-SUBMIT
```
Comment (avimehta) [inline extensions/amp-analytics/0.1/cid-impl.js]: planing to remove this line?

File: spec/amp-var-substitutions.md [hunk 2]
```diff
@@ -445,9 +445,9 @@ Provides a per document-source-origin (the origin of the website where you publi
 
 You can pass the following arguments into the Client ID variable like a function. Spaces between arguments and values are not allowed.
 
-  - `cid scope` (Required): The name of the fallback cookie when the document
-    is not served by an AMP proxy.
+  - `cid scope` (Required): The namespace for the CID.
```
Comment (avimehta) [inline spec/amp-var-substitutions.md]: would be better to expand CID to clientId

**Comment by avimehta** [comment]
oh, and one if the test fails.

File: extensions/amp-analytics/0.1/cid-impl.js (hunk 1 above)
Comment (lannka) [inline extensions/amp-analytics/0.1/cid-impl.js]: yes. added to catch any code that are passing `string` . will remove once all tests pass

File: spec/amp-var-substitutions.md (hunk 2 above)
Comment (lannka) [inline spec/amp-var-substitutions.md]: done

**Pull request merged**